experimental/torchprep/torchprep/main.py

```diff
@@ -0,0 +1,193 @@
+import typer
+import torch
+from typing import List, Union
+import time
+import os
+from enum import Enum
+from pathlib import Path
+import math
+from tqdm import tqdm
+import torch.fx as fx
+import torch.nn.utils.prune
+
+
+app = typer.Typer()
+
+class Precision(Enum):
+    int8 = "int8"
+    float16 = "float16"
+
+class Device(str, Enum):
+    cpu = "cpu"
+    gpu = "gpu"
+
+class TensorType(str, Enum):
+    float = "float"
+    int   = "int"
+    long  = "long"
+
+class Architecture(str, Enum):
+    ipex = "ipex"
+    tensorrt = "tensorrt"
+    fastertransformer = "fastertransformer"
+
+
+@app.command()
+def distill(model_path : Path, device : Device = Device.cpu, parameter_scaling : int = 2, layer_scaling : int = None, profile : List[int] = None) -> torch.nn.Module:
+    """
+    [Coming soon]: Create a smaller student model by setting a distillation ratio and teach it how to behave exactly like your existing model
+    """
+    print(f"Coming soon")
+    print("See this notebook for more information https://colab.research.google.com/drive/1RzQtprrHx8PokLQsFiQPAKzfn_DiTpDN?usp=sharing")
+
+@app.command()
+def prune(model_path : Path, prune_amount : float = typer.Option(default=0.3, help=" 0 < prune_amount < 1 Percentage of connections to prune"), device : Device = Device.cpu) -> torch.nn.Module:
+    """
+    Zero out small model weights using l1 norm
+    """
+    model = load_model(model_path, device)
+    
+    for name, module in model.named_modules():
+        if isinstance(module, torch.nn.Conv2d) or isinstance(module, torch.nn.Linear) or isinstance(module, torch.nn.LSTM):
+            torch.nn.utils.prune.l1_unstructured(module, "weight", prune_amount)
+    
+    torch.save(model, 'pruned_model.pt')
+    print("Saved prune model pruned_model.pt")
+    return model
+
+
+
+@app.command()
+def fuse(model_path : Path, device : Device = Device.cpu,input_shape : str = typer.Option(default=None, help="Comma seperated input tensor shape e.g [64,3,7,7]"), input_type : TensorType = TensorType.float) -> torch.nn.Module:
```

Comment:
@msaroufim we might be able to use a more descriptive name here, similar to optimize_for_inference?